File: LC_118.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> triangle = new ArrayList<List<Integer>>();
        
        if (numRows == 0)
            return triangle;
        
        triangle.add(new ArrayList<>());
        triangle.get(0).add(1);
        
        for (int i = 1; i < numRows; i++) {
            List<Integer> newRow = new ArrayList<>();
            List<Integer> prevRow = triangle.get(i - 1);
            
            newRow.add(1);
            
            for (int j = 1; j < i; j++)
                newRow.add(prevRow.get(j - 1) + prevRow.get(j));
            
            newRow.add(1);
            
            triangle.add(newRow);
        }
        
        return triangle;
    }
}

File: LC_118.py
# 118. Pascal's Triangle

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = [[1]]
        
        for i in range(numRows - 1):
            row = []
            row.append(1)
            for j in range(len(result[-1]) - 1):
                row.append(result[-1][j] + result[-1][j + 1])
            row.append(1)
            result.append(row)
        
        return result
    